File: Machine Learning A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_liner_regression.R
#import dataset
dataset = read.csv('Salary_Data.csv')

#Split dataset
library(caTools)
set.seed(123)
split = sample.split(Y = dataset$Salary, SplitRatio = 0.8)

train_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Fitting LinerRegression Model
lr = lm(formula = Salary ~ YearsExperience,
        data = train_set)

#Prediction on test set
y_pred = predict(lr,test_set)

#Visualization
library(ggplot2)
#Traingin set
ggplot() +
  geom_point(aes(x = train_set$YearsExperience, y = train_set$Salary),
             colour = 'red') +
  geom_line(aes(x = train_set$YearsExperience, y = predict(lr,train_set)),
            colour = 'blue') +
  ggtitle("Salary vs Experience(Training set)") +
  xlab("Years of Experience") +
  ylab("Salary in USD")

#Test set
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = train_set$YearsExperience, y = predict(lr,train_set)),
            colour = 'blue') +
  ggtitle("Salary vs Experience(Testing set)") +
  xlab("Years of Experience") +
  ylab("Salary in USD")

File: Machine Learning A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_liner_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Oct  8 00:20:38 2019

@author: diptu
"""
#import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import datasetx
df = pd.read_csv('Salary_Data.csv')
X = df.iloc[:, :-1].values
y = df.iloc[:,1].values

#Split data into train and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

#Fitting Simple Linear Regression
from sklearn.linear_model import LinearRegression
lr = LinearRegression()
lr.fit(X_train, y_train)

#Predict test result
y_pred = lr.predict(X_test)

#Visualization the training set result
plt.scatter(X_train, y_train, color='red')
plt.plot(X_train, lr.predict(X_train), color='blue')
plt.title('Salary vs Experience (Train set)')
plt.xlabel('Salary in USD')
plt.ylabel('Experience(year)')
plt.show()


#Visualization the test set result
plt.scatter(X_test, y_test, color='green')
plt.plot(X_train, lr.predict(X_train), color='blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Salary in USD')
plt.ylabel('Experience(year)')
plt.show()






